quint/src/parsing/ToIrListener.ts

```diff
@@ -764,13 +764,33 @@ export class ToIrListener implements QuintListener {
   }
 
   // record constructor, e.g., { name: "igor", year: 2021 }
+  private checkDuplicateFields(pairs: QuintEx[][], ctx: p.RecordContext): boolean {
+    const seenFields = new Set<string>()
+    for (const pair of pairs) {
+      const [key] = pair
+      if (key.kind === 'str') {
+        if (seenFields.has(key.value)) {
+          const id = this.getId(ctx)
+          this.errors.push(duplicateRecordFieldError(id, key.value))
+          return true
+        }
+        seenFields.add(key.value)
+      }
+    }
+    return false
+  }
+
   exitRecord(ctx: p.RecordContext) {
     const elems = popMany(this.exprStack, ctx.recElem().length, this.undefinedExpr(ctx))
     const spreads = elems.filter(e => e.kind === 'app' && e.args.length === 1)
     const pairs = elems.map(e => (e.kind === 'app' && e.args.length === 2 ? e.args : [])).filter(es => es.length > 0)
 
     if (spreads.length === 0) {
-      // { field1: value1, field2: value2 }
+      // Check for duplicate fields before constructing the record
```

Comment:
We need this check even if spreads are present:
```
>>> val a = { foo: 1, bar: 2 }

>>> { ...a, bar: 2, bar: 3 }
{ bar: 3, foo: 1 }
```

(this should also result in an error)

Reply:
Seems like you might have forgotten to update the import